installer/pkg/config/v1/validation.go

```diff
@@ -150,5 +150,9 @@ func (v version) ClusterValidation(rcfg interface{}) cluster.ValidationChecks {
 		}
 	}
 
+	if cfg.SSHGatewayHostKey != nil {
+		secretName := cfg.SSHGatewayHostKey.Name
+		res = append(res, cluster.CheckSecret(secretName))
```

Comment:
We might want to check that there's a key present at all, i.e. the secret contains at least entry in `data`. Bonus points for validating that that's a valid key :)